lib/learn_to_rank/reranker.rb

```diff
@@ -0,0 +1,35 @@
+require "learn_to_rank/feature_sets"
+require "learn_to_rank/ranker"
+
+module LearnToRank
+  class Reranker
+    # Reranker re-orders elasticsearch results using a pre-trained model
+    def rerank(es_results: [], count: DEFAULT_COUNT)
+      feature_sets = FeatureSets.new.call(es_results)
+      new_scores   = Ranker.new(feature_sets).ranks
+
+      reorder_results(es_results, new_scores)
+    end
+
+  private
+
+    DEFAULT_COUNT = 20
+    MAX_MODEL_BOOST = 5
+
+    def reorder_results(search_results, new_scores)
+      search_results
+        .map
+        .with_index { |result, index|
+          m_score = [new_scores[index], MAX_MODEL_BOOST].min
+          es_score = result.fetch("_score", 0)
+          result.merge(
+            "model_score" => m_score,
+            "original_rank" => index + 1,
+            "combined_score" => m_score * es_score,
```

Comment:
I'm not certain about multiplying the scores.  Does the tensorflow model know it's producing a multiplicative factor, or does it think it's producing entirely new scores?

Reply:
The model assumes it is producing new scores (not something to be multiplied). This was chosen so we could treat the model as a booster while we're not certain about its results.

I agree that it shouldn't be a multiplicative factor. We should just the new scores.

It has one benefit though, that best bets continue to work. But I think we should replace the best bet process with generating new relevancy judgements for a query.

Reply:
I'm not sure the new ranks we're getting are particularly valid: new scores and boosting factors to existing scores are very different things.

We can keep best bets working by using the original score if it's greater than whatever the smallest best bet score is.